packages/@aws-cdk/aws-servicecatalog/lib/common.ts

```diff
@@ -18,4 +23,73 @@ export enum MessageLanguage {
    * Chinese
    */
   ZH = 'zh'
-}
\ No newline at end of file
+}
+
+/**
+ * The types of cloudFormationTemplates for product versions.
+ * Used to determine the source of a cloudFormationTemplate and apply logic accordingly
+ */
+export enum TemplateType {
+  /**
+   * AssetTemplate
+   */
+  ASSET = 'AssetTemplate',
+
+  /**
+   * UrlTemplate
+   */
+  URL = 'UrlTemplate',
+
+  /**
+   * ProductStackTemplate
+   */
+  PRODUCT_STACK = 'ProductStackTemplate',
+
+  /**
+   * ProductStackContextTemplate
+   */
+  PRODUCT_STACK_CONTEXT = 'ProductStackContextTemplate'
+}
+
+/**
+   * The strategy to use for a ProductStack deployment.
+   * Determines how a productVersion is saved and deployed.
+   */
+export enum VersioningStrategy {
+  /**
+   * Default Strategy for ProductStack deployment.
+   * This strategy will overwrite existing versions when deployed.
```

Comment:
I don't understand these comments if a `VersioningStrategy` is defined for each version. This comment makes me think that this is defined per product; if one version has a retain strategy, and another version has the overwrite strategy, how do we decide which versions are retained? Are there versions of versions?

Reply:
I see. Effectively we have a product version A, B, C and then A, B and C can each have versions. I'd prefer we call product versions something other than versions, if we're also going to call the versions of versions "versions".

Reply:
I think the mistake here is the `VersioningStrategy` name, as it re-uses the term "version" to mean something else than does it now.

How about calling it `RetentionStrategy` instead, with two values: `OVERRIDE` (the default), and `RETAIN`.

Reply:
If we do this then let's call it `OVERWRITE`. But I favor a different type of API in general.